src/retry.rs

```diff
@@ -0,0 +1,25 @@
+use tokio_retry::strategy::{FibonacciBackoff, jitter};
+use std::future::Future;
+use std::{
+  iter::{Take, Map},
+  time::Duration
+};
+
+fn retry_strategy(ms: u64, attempts: usize) -> Take<Map<FibonacciBackoff, fn(Duration) -> Duration>>{
```

Comment:
```suggestion
fn retry_strategy(base: Duration, attempts: usize) -> impl Iterator<Item = Duration> {
```

Reply:
The function itself probably isn't necessary. It's very simple and it's used only once. It only bloats the code. I'd put the contents directly into `with_retry`.